### ISSUE TITLE
Mettre à jour les tests d'intégration continue

### ISSUE THREAD

**Issue by erwan-simon-CATIE** [pixel-ports/pas_modelling#5]

**Comment by erwan-simon-CATIE** [comment]
Il se passe des choses étranges avec les yml de test. Alors que ces vieux fichiers de test périmés sont supprimés (apparemment partout, cf recherche sur tout le dépôt), l'un des deux semble toujours actif. Il est toujours présenté dans la section "Action".
![image](https://user-images.githubusercontent.com/60698075/87848259-bd278480-c8de-11ea-947a-a4f389f769a4.png)

**Comment by rqueraud** [comment]
Tu en as un ici : https://github.com/pixel-ports/pas_modelling/blob/feature/MaJ_tests/.github/workflows/develop_minimal_test.yml.

Mais c'est dommage de les enlever, non ? On ne pourrait pas les mettre à jour plutôt ?

**Comment by erwan-simon-CATIE** [comment]
Ba celui-là c'est "develop_minimal_test.yml", cad un test mis à jour, lui je veux qu'il existe :).

Là j'avais des échecs avec les deux anciens qui étaient déclenchés je ne sais pas prk. Deux éléments d'info complémentaires :
1) l'affichage dans Action est principalement lié à la branche par défaut (cad master pr nous). C'est plus compliqué, mais cette notion de branche officieusement prioritaire existe.
2) develop_minimal_test.yml est pas tout à fait un nouveau fichier. C'est un des deux anciens renommé et contenu changé (en local) avant de push. Mais théoriquement, c'est équivalent à 2 sup et 1 ajout.

(NB je serais étonné que la première tentative de develop_minimal_test.yml fonctionne, mais c'est une autre question)

**Comment by erwan-simon-CATIE** [comment]
Propositions de modif (cf le readme) les tests seront par branche (on ne teste pas les même choses).
- Master à ces tests qui portent sur une image docker
- Dev à ces tests qui portent sur son exécution sans erreur, plus les tests des branches feature qui ont été merge
- Chaque branche feature devrait avoir n tests particuliers

https://github.com/pixel-ports/pas_modelling/tree/feature/MaJ_tests/.github/workflows

**Comment by rqueraud** [comment]
> Ba celui-là c'est "develop_minimal_test.yml", cad un test mis à jour, lui je veux qu'il existe :).
> 
> Là j'avais des échecs avec les deux anciens qui étaient déclenchés je ne sais pas prk. Deux éléments d'info complémentaires :
> 
> 1. l'affichage dans Action est principalement lié à la branche par défaut (cad master pr nous). C'est plus compliqué, mais cette notion de branche officieusement prioritaire existe.
> 2. develop_minimal_test.yml est pas tout à fait un nouveau fichier. C'est un des deux anciens renommé et contenu changé (en local) avant de push. Mais théoriquement, c'est équivalent à 2 sup et 1 ajout.
> 
> (NB je serais étonné que la première tentative de develop_minimal_test.yml fonctionne, mais c'est une autre question)

C'est lequel qui te gène alors ? Le `FunkyPAS.yml` ?

Parce que quand je regarde les commits liés aux actions qui l'ont déclenché, le fichier est bien présent.

**Comment by rqueraud** [comment]
> Propositions de modif (cf le readme) les tests seront par branche (on ne teste pas les même choses).
> 
> * Master à ces tests qui portent sur une image docker
> * Dev à ces tests qui portent sur son exécution sans erreur, plus les tests des branches feature qui ont été merge
> * Chaque branche feature devrait avoir n tests particuliers
> 
> https://github.com/pixel-ports/pas_modelling/tree/feature/MaJ_tests/.github/workflows

Quel est l'intérêt de séparer les tests ? On pourrait tout executer à chaque fois, non ? 

Séparer les tests, j'ai peur que ce soit + une source de problème qu'autre chose, car on va s'y perdre.

**Comment by erwan-simon-CATIE** [comment]
![image](https://user-images.githubusercontent.com/60698075/87917266-fe519d00-ca74-11ea-8db6-c2a074709cd3.png)

**Comment by erwan-simon-CATIE** [comment]
Je ne comprends pas l'interface Github. Dans Action, on voit encore FunkyPAS.yml, je ne sais pas prk.
MAIS il n'est plus actif (le dernier push où il réagit était il y à 3 j, contre 2 pr develop_minimal_test.yml ). Donc à priori, il s'agirait juste de le virer pr faire propre et comprendre le prk du comment. Mais c'est pas des tâches critiques.
Le truc important serait d'édit develop_minimal_test.yml pr qu'il fonctionne.

![image](https://user-images.githubusercontent.com/60698075/87917381-322cc280-ca75-11ea-8dea-b85d2e1da277.png)

**Comment by rqueraud** [comment]
C'est normal qu'il apparaisse encore ici. Il est référencé par des anciens commits.

**Comment by erwan-simon-CATIE** [comment]
L'autre ancien test ne l'était il pas ? Il était pourtant push aussi, et avait été trigger...

**Comment by rqueraud** [comment]
Ah t'as raison, c'est bizarre du coup qu'il n'apparaisse pas non plus le `pas_library.yml`. Surtout que les premières executions le mentionnent bien.

**Comment by erwan-simon-CATIE** [comment]
> Quel est l'intérêt de séparer les tests ? On pourrait tout executer à chaque fois, non ?

Ba déjà il y a l'idée que sur master on publie une image docker alors que sur dev on garde des fichiers (ce n'est qu'une proposition, mais si acceptée alors on doit séparer les tests). Après, si je suis en train de dév une feature X, le test qui me permet de savoir que mon code fonctionne, puis qu'une fois merge sur dev, c'est pas cassé lorsque tu merges tu code pr la feature Y...  c'est un autre test que celui que t'as dév pr la feature Y. Du coup pr moi c'est n tests à stacker sur Develop (à chaque fois qu'il y a un merge d'une nouvelle feature). Non ?

> Séparer les tests, j'ai peur que ce soit + une source de problème qu'autre chose, car on va s'y perdre.

A ce stade pr moi, séparer  = trigger différents tests pr différents types de push (principalement sur la base de la branche). Après, on peut soit faire un seul fichier .yml qui contient tout l'arbre des tests (tu peux faire des tas de choses, des maps, des exclusions....), soit n fichiers, chacun simple avec un périmètre réduit. Ce genre de détail, avis neutre pour ma part

**Comment by rqueraud** [comment]
* Pour restreindre la publication de l'image docker quand on est sur la branche master tu as [ça](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags). Ça te permet de choisir sur quelle branche tu executes ton job.
* Effectivement, stacker les tests c'est ce qu'il faut faire. Et donc ta branche develop devra executer les tests de X + les tests de Y. On est d'accord, j'avais mal compris ton premier message :). Et donc si tu tires ensuite une nouvelle feature branche Z, elle executera les tests de X, Y et Z pour vérifier qu'elle n'a rien cassé des précédentes features ajoutées.

**Comment by erwan-simon-CATIE** [comment]
Oui, actuellement, le test ne trigger que sur develop
`on: 
    push:
        branches:
            - develop`

Mais sur le reste, mon "code" de test n'est pas viable. Je galère pr comprendre la synthaxe. Tu veux bien y jetter un coup d'oeil ? Il est probable que tu puisses corriger le truc en 2 min.

**Comment by rqueraud** [comment]
De quel code de test tu parles, du `PAS_builder.py` ? Sur la branche `MaJ_tests` ?

Parce que si je run la commande `python3 ./PAS_builder.py --pipeline=local_PAS_instance, --OT_input=local_PAS_instance` effectivement j'ai un fichier qui est créé avec ça dedans : 

```json
{
    "LOGS": [
        "==== main  ====",
        "Pipeline: local_PAS_instance,",
        "Converting OT_input to dict: Failled. Error: Expecting value: line 1 column 1 (char 0)",
        "Loading Settings: Failled. Error: 'local_PAS_instance,'",
        "PAS modelling closing"
    ],
    "ACTIVITIES": null,
    "PORT'S PARAMETERS": null,
    "SETTINGS": null
}
```

**Comment by erwan-simon-CATIE** [comment]
Oui, certains trucs n'étaient pas synchro entre les branche develope et MaJ_test.
- le test se déroulait correctement
- il détectait une erreur "réelle" dans le code (corrigée après rebase entre les branches)

**Comment by erwan-simon-CATIE** [comment]
Pr le fait que FunkyPAS.yml apparaisse comme un famtome, c'est p'tre un artefact de ce genre là
https://github.community/t/how-can-i-edit-an-existing-action/18111/3

C'est pas prioritaire, comme ce "test" ne se déclenche pas, on peut voir plus tard
